include/itkImageToTorchImageFilter.h

```diff
@@ -0,0 +1,106 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkImageToTorchImageFilter_h
+#define itkImageToTorchImageFilter_h
+
+#include "itkImageToImageFilter.h"
+#include "itkImageRegionIterator.h"
+#include "itkImageRegionConstIterator.h"
+
+#include "itkTorchImage.h"
+
+namespace itk
+{
+
+/** \class ImageToTorchImageFilter
+ *
+ * \brief Converts an itk::Image to an itk::TorchImage.
+ *
+ * Conversions between Image and TorchImage permit pipelines with
+ * steps for either kind of image.
+ *
+ * \ingroup PyTorch
+ *
+ */
+template< typename TInputImage >
+class ITK_TEMPLATE_EXPORT ImageToTorchImageFilter
+  : public ImageToImageFilter< TInputImage, TorchImage< typename TInputImage::PixelType, TInputImage::ImageDimension > >
```

Comment:
We want the ability to use the filter, *Input, Output* with all the template arguments:

- *Image, TorchImage*
- *TorchImage, Image*
- *TorchImage, TorchImage*

without requiring a change to the class used. Also, we want to re-use the CPU image information metadata transformation logic, and only write one Torch implementation of the `GenerateData`. This implementation requires client code to support three different classes, requires re-implementation of the metadata transformation logic, and would require multiple copies of the `GenerateData` logic. The design pattern should follow that of `GPUImageToImageFilter`:

 https://github.com/InsightSoftwareConsortium/ITK/blob/4f6fc719bd1e51cbd8f12f15ef3286a18e006996/Modules/Core/GPUCommon/include/itkGPUImageToImageFilter.h#L26-L41